pkg/distributor/distributor_test.go

```diff
@@ -115,6 +115,53 @@ func Test_SortLabelsOnPush(t *testing.T) {
 	require.Equal(t, `{a="b", buzz="f"}`, ingester.pushed[0].Streams[0].Labels)
 }
 
+func Test_TruncateLogLines(t *testing.T) {
```

Comment:
One last thing from my side:
Can you please add a benchmark for this functionality? We should verify that this doesn't greatly increase the ingestion latency.

Reply:
Benchmarks added and it was a good thing too. Calls to `validation.Mutated{Sample,Bytes}.WithLabelValues` are shockingly expensive. Moving them outside the loop reduced allocs 100x

Reply:
Changes that brought allocs down were editing the entry line in place and moving metrics reporting outside of the truncation loop.

Before Changes:
```
goos: linux
goarch: amd64
pkg: github.com/grafana/loki/pkg/distributor
cpu: 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
Benchmark_SortLabelsOnPush-8                            16393114                61.31 ns/op           16 B/op          1 allocs/op
Benchmark_Push-8                                            1105           1094502 ns/op            1978 B/op         37 allocs/op
Benchmark_PushWithLineTruncation-8                           914           1301114 ns/op            8999 B/op        255 allocs/op
Benchmark_PushWithLineTruncationWithIndicator-8              910           1429702 ns/op           16009 B/op        366 allocs/op
PASS
```

After:
```
goos: linux
goarch: amd64
pkg: github.com/grafana/loki/pkg/distributor
cpu: 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
Benchmark_SortLabelsOnPush-8                            19185693                61.86 ns/op           16 B/op          1 allocs/op
Benchmark_Push-8                                            1185           1033126 ns/op            1968 B/op         37 allocs/op
Benchmark_PushWithLineTruncation-8                           921           1366202 ns/op            2006 B/op         39 allocs/op
Benchmark_PushWithLineTruncationWithIndicator-8              856           1361395 ns/op            9499 B/op        155 allocs/op
PASS
```

As expected, adding the Indicator bumps allocs a fair bit.

Reply:
Thanks for adding these

I like that the truncation barely adds any alloc overhead. With the indicator, I think it's ok to alloc a bit more as it's user-configurable behaviour and disabled by default. The bytes per op, I'm curious about.

@cyriltovena what do you think?